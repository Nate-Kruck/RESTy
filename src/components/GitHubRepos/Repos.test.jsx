import React from 'react';
import {
  render,
  cleanup,
  screen
} from '@testing-library/react';
import RepoList from '../GitHubRepos/Repos';

describe('RepoList Testing', () => {
  afterEach(() => cleanup());

  it('renders a RepoList', () => {
    const repos = [
      {
        url: 'https://api.github/users/nate-kruck',
        name: 'restyAPI-1',
        login: 'nate-kruck',
      },
      {
        url: 'https://api.github/users/nate-kruck',
        name: 'restyAPI-2',
        login: 'nate-kruck',
      },
      {
        url: 'https://api.github/users/nate-kruck',
        name: 'restyAPI-3',
        login: 'nate-kruck',
      }
    ];
    const { asFragment } = render(<RepoList repos={repos} />);

    const allLogins = screen.getAllByText('nate-kruck');

    expect(allLogins.length).toEqual(3);

    expect(asFragment()).toMatchSnapshot();
  });
});
