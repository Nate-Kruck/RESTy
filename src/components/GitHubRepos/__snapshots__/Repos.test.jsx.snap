// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList Testing renders a RepoList 1`] = `
<DocumentFragment>
  <div>
    <ul>
      <div>
        <li>
          <a
            href="https://api.github/users/nate-kruck"
            rel="noreferrer"
            target="_blank"
          >
            <h1
              data-testid="repo"
            >
              restyAPI-1
            </h1>
          </a>
        </li>
        <p>
          nate-kruck
        </p>
      </div>
      <div>
        <li>
          <a
            href="https://api.github/users/nate-kruck"
            rel="noreferrer"
            target="_blank"
          >
            <h1
              data-testid="repo"
            >
              restyAPI-2
            </h1>
          </a>
        </li>
        <p>
          nate-kruck
        </p>
      </div>
      <div>
        <li>
          <a
            href="https://api.github/users/nate-kruck"
            rel="noreferrer"
            target="_blank"
          >
            <h1
              data-testid="repo"
            >
              restyAPI-3
            </h1>
          </a>
        </li>
        <p>
          nate-kruck
        </p>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
